ctk/common/src/main/kotlin/org/codice/compliance/verification/binding/RedirectBindingVerifier.kt

```diff
@@ -260,16 +279,33 @@ class RedirectBindingVerifier(private val response: IdpRedirectResponseDecorator
     @Suppress("ComplexMethod" /* Complexity due to nested `when` is acceptable */)
     private fun decodeAndVerifyErrorResponse() {
         val samlResponse = response.samlResponse
-        val samlEncoding = response.samlEncoding
-        val decodedMessage: String
+
+        // Need to url decode SAMLEncoding first to check the encoding method uri
+        val samlEncoding = response.samlEncoding?.let {
+            try {
+                URLDecoder.decode(it, StandardCharsets.UTF_8.name())
+            } catch (e: UnsupportedEncodingException) {
+                throw SAMLComplianceException.create(SAMLBindings_3_4_4_1_c,
+                        message = "Could not url decode the SAMLEncoding parameter.",
+                        cause = e)
+            }
+        }
+
+        samlEncoding?.let {
+            if (StringUtils.isBlank(it) || !URI.create(it).isAbsolute)
+                throw SAMLComplianceException.create(SAMLCore_1_3_2_a,
+                        SAMLBindings_3_4_4_1_e,
+                        message = "The URI value of SAMLEncoding [$it] is " +
+                                "invalid.")
+        }
 
         /**
          * A query string parameter named SAMLEncoding is reserved to identify the encoding
          * mechanism used. If this parameter is omitted, then the value is assumed to be
          * urn:oasis:names:tc:SAML:2.0:bindings:URL-Encoding:DEFLATE.
          */
-        decodedMessage = if (samlEncoding == null ||
-                samlEncoding.equals("urn:oasis:names:tc:SAML:2.0:bindings:URL-Encoding:DEFLATE")) {
+        val decodedMessage = if (samlEncoding == null ||
```

Comment:
✏️ samlEncoding won't be null. StringUtils.isBlank checks for that

Reply:
No longer an issue after a little refactor I did.